src/library_pthread.js

```diff
@@ -416,6 +422,8 @@ var LibraryPThread = {
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 20) >> 2, threadParams.schedPolicy);
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 24) >> 2, threadParams.schedPrio);
 
+    Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.locale }}}) >> 2, __global_locale_ptr);
```

Comment:
Oh I see, the propagation to other threads is done here. It would be better to directly have the data live in the shared heap so this would not be needed. Better yet, that could avoid the need for the Emscripten specific `__emscripten_setup_pthread_data()` altogether?